tests/attestation/test.py

```diff
@@ -0,0 +1,255 @@
+2Naz5m2B+O+vjsC060d37jY9LFeuOP4Meri8qgfi2S5kKqg/aF6aPtuAZQVR7u3K
+FYXP59XmJgtcog05gmI0T/OitLhuzVvpZcLph0odh/1IPXqx3+MnjD97A7fXpqGd
+/y8KxX7jksTEzAOgbKAeam3lm+3yKIcTYMlsRMXPcjNbIvmsBykD//xSniusuHBk
+gnlENEWx1UcbQQrs+gVDkuVPhsnzIRNgYvM48Y+7LGiJYnrmE8xcrexekBxrva2V
+9TJQqnN3Q53kt5viQi3+gCfmkwC0F0tirIZbLkXPrPwzZ0M9eNxhIySb2npJfgnq
+z55I0u33wh4r0ZNQeTGfw03MBUtyuzGesGkcw+loqMaq1qR4tjGbPYxCvpCq7+Og
+pCCoMNit2uLo9M18fHz10lOMT8nWAUvRZFzteXCm+7PHdYPlmQwUw3LvenJ/ILXo
+QPHfbkH0CyPfhl1jWhJFZasCAwEAAQ==
+-----END PUBLIC KEY-----
+""".encode(),
+)
+
+
+def get_certificate_chain(certificate_chain: bytes) -> Tuple[x509.Certificate, ...]:
+    certificate_chain_json = json.loads(
+        base64.b64decode(certificate_chain).decode())
+    return tuple(
+        [
+            x509.load_pem_x509_certificate(cert.encode())
+            for cert in [
+                certificate_chain_json["vcekCert"],
+                *re.findall(
+                    f"-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----",
+                    certificate_chain_json["certificateChain"],
+                    flags=re.DOTALL,
+                ),
+            ]
+        ]
+    )
+
+
+def cert_signed_other_cert(
+    parent_cert: x509.Certificate,
+    child_cert: x509.Certificate,
+) -> bool:
+    try:
+        parent_cert.public_key().verify(  # type: ignore
+            child_cert.signature,
+            child_cert.tbs_certificate_bytes,
+            PSS(
+                mgf=MGF1(SHA384()),
+                salt_length=SHA384.digest_size,  # type: ignore
+            ),  # type: ignore
+            SHA384(),  # type: ignore
+        )
+        return True
+
+    except InvalidSignature:
+        return False
+
+
+def validate_attestation(
+    attestation_bytes: bytes,
+    certificate_chain: bytes,
+    expected_report_data: str,
+) -> bool:
+    (
+        version,
+        guest_svn,
+        policy,
+        family_id,
+        image_id,
+        vmpl,
+        signature_algorithm,
+        current_tcb,
+        platform_info,
+        author_key,
+        report_data,
+        measurement,
+        host_data,
+        id_key_digest,
+        author_key_digest,
+        report_id,
+        report_id_ma,
+        reported_tcb,
+        chip_id,
+        committed_tcb,
+        current_build,
+        current_minor,
+        current_major,
+        committed_build,
+        committed_minor,
+        committed_major,
+        launch_tcb,
+        signature,
+    ) = struct.unpack_from(f"<{SNP_REPORT_STRUCTURE}", attestation_bytes, 0)
+
+    # Validate report was generated based on the provided report data
+    # (i.e. This is the report we requested)
+    seen_report_data = report_data.rstrip(b"\x00").decode()
+    print(f"Checking that report data in the attestation: {seen_report_data}")
+    print(f"Matches the report data we provided: {expected_report_data}")
+    assert seen_report_data == expected_report_data
```

Comment:
Logging would be cleaner if there was an error message along with the assert rather than print statements before and after the assert

e.g.
`assert seen_report_data == expected_report_data, "Expected report data does not match what was seen"`

These could also be f-strings if you want to have both pieces of data printed out

Reply:
I agree except I think it's nice to see the logs even in the success case, which you don't get with the following message